flax/linen/module.py

```diff
@@ -344,6 +359,7 @@ def __init_subclass__(cls):
     cls._wrap_module_methods()
     # Set empty class defaults.
     cls._state = _uninitialized_module_internal_state
+    cls._setup_called = False
```

Comment:
I think this belongs in ModuleInternalState

Reply:
Actually this is interesting, and correct me if I'm wrong, but I -think- this actually should be object-local, rather than state that is copied across transform boundaries.  We don't want `_setup_called` True inside a cloned object to copy to the outside object, since it in fact has _not_ had `setup` called. (For instance __call__ has been called under a transform, but now a user triggers `__getattr__`, or in the case of multiple methods - a second method needs to rebuild the object inside its own transform context).  We could still stick it inside `ModuleInternalState` but it would require manually writing these  scoping rules.

Reply:
I would put it in ModuleInternalState regardless. Just makes it easier to handle when you have less magic attributes especially with all this setattr getattr logic.

Reply:
Yeah you're right, all runtime state should be kept together and the transform-boundary behavior made explicit - fixed.